multiplexer/internal/utils.go

```diff
@@ -60,14 +60,7 @@ func GetGenesisVersion(genesisPath string) (GenesisVersion, error) {
 		}
 	}
 
-	var v2 genesisDocv2
-	if err := json.Unmarshal(genDoc, &v2); err == nil {
-		if v2.Consensus.Params.Version.App != "" {
-			return GenesisVersion2, nil
-		}
-	}
-
-	return 0, errors.New("failed to determine genesis version")
+	return GenesisVersion2, nil
```

Comment:
note to self, revisit this as we might be able to delete other things if we limit this check

Reply:
I preferred the more explicit error if it couldn't determine the genesis version. Why remove that?

Reply:
the genesis parsing logic would produce false negatives so had to be removed to get a v4 genesis to work (at least for me when creating this. perhaps it has since been fixed)

Reply:
yeah any v4 genesis

>  this will now silently return genesis version v2 when it can't deserialize a genesis.json.

then the normal genesis deserialization will fail and return a more appropriate error. The only thing we need to do with this function is check if its an old genesis, not also confirm that its a new genesis